### DISCUSSION TITLE
Camera Setup

### DISCUSSION THREAD

**Discussion by KSumwalt** [andrei-tatar/node-red-contrib-smartnora#156] [Q&A]
I have IP Cameras which I would like to use with the Camera node but am having problems configuring them to display.  I can Cast using a different application.  I have tried a few different methods with both internal IPs and an external URL as a subdomain.  I can view the external URL in a browser on my phone with the WiFi turned off.  Here is a sample URL:  http://bi1.sumwalt.net/  OR http://gs1.sumwalt.net/  These use different internal methods. The first is a Blue Iris instance and the second is the feed from a one of the 4 cameras on a Grandstream GXV3504.  Both show the same camera.  The Grandstream unit converts it from analog to digital.  The Grandstream feed is MJPEG using the internal URL from https://www.ispyconnect.com/camera/grandstream.

Ideally I would like to use the Grandstream feed as I am only testing Blue Iris for additional streaming formats and do not intend to use it unless it is my only option.  I am open to using some other software internally if needed.

I have placed the above URLs in different fields in the Camera node, may have missed one, but each time the best I get is a screen loading the camera showing a Play button and "Playing Smart Home Camera" followed by a black screen with white "Smart Hoime Camera" showing.  Any thoughts?

**Comment by andrei-tatar** [comment]
@KSumwalt I don't think MJPEG is supported: https://developers.google.com/assistant/smarthome/traits/camerastream#device-attributes

I also have some cameras that expose an RTSP endpoint but I didn't have time to check how to integrate them with Google Home. But I'm pretty sure I need something like this as a middleman: https://github.com/deepch/RTSPtoWeb

**Comment by patrik-malina** [comment]
Hi all. I'm using this https://github.com/AlexxIT/go2rtc
It could restream several codecs and protocols.
Managed to stream to Google Nest Hub HLS and MP4 stream, but its very slow and laggy.

Fastest is WebRTC, but I cannot manage to work it.

If anyone is willing to try and share solution, you are welcome...
